lib/Common/Memory/Recycler.h

```diff
@@ -758,7 +764,27 @@ class Recycler
     uint collectionFinishReason;
 #endif
 
-    CollectionState collectionState;
+    ObservableCollectionState collectionState;
+
+    inline void SetCollectionState(CollectionState newState)
+    {
+        this->collectionState = newState;
+    }
+
+    virtual void CollectionStateChanged(CollectionState newState, CollectionState oldState)
+    {
+#ifdef ENABLE_BASIC_TELEMETRY
+        if (this->collectionState == CollectionState::CollectionStateNotCollecting && newState != CollectionState::CollectionStateNotCollecting && newState != CollectionState::Collection_PreCollection)
+        {
+            this->GetRecyclerTelemetryInfo().StartPass();
+        }
+        else if (this->collectionState != CollectionState::CollectionStateNotCollecting && this->collectionState != CollectionState::Collection_PreCollection && newState == CollectionState::CollectionStateNotCollecting)
```

Comment:
We can go from preCollection to NotCollecting?

Reply:
looks like it:  https://github.com/Microsoft/ChakraCore/blob/master/lib/Common/Memory/Recycler.cpp#L3861-L3864